lib/combine_pdf/renderer.rb

```diff
@@ -84,7 +84,7 @@ def format_name_to_pdf(object)
     def format_array_to_pdf(object)
       # An array shall be written as a sequence of objects enclosed in SQUARE BRACKETS (using LEFT SQUARE BRACKET (5Bh) and RIGHT SQUARE BRACKET (5Dh)).
       # EXAMPLE [549 3.14 false (Ralph) /SomeName]
-      ('[' + (object.collect { |item| object_to_pdf(item) }).join(' ') + ']').force_encoding(Encoding::ASCII_8BIT)
+      ('[' + (object.collect { |item| object_to_pdf(item) }).join(' ') + ']').dup.force_encoding(Encoding::ASCII_8BIT)
```

Comment:
(again) This dup is redundant and will use up a bunch of memory. The dynamic String used here should be guaranteed to be mutable.